.github/workflows/cicd_tests.yml

```
@@ -0,0 +1,319 @@
+  # quick tests for pull requests and the releasing branches
+  push:
+    branches:
+      - dev
+      - main
+      - releasing/*
+    paths-ignore:  # skip if only docs are modified
+      - '**.md'
+      - '**.rst'
+      - 'docs/**'
+  pull_request:
+    head_ref-ignore:
```

actionlint (1.7.12)

[error] unexpected key "head_ref-ignore" for "pull_request" section. expected one of "branches", "branches-ignore", "paths", "paths-ignore", "tags", "tags-ignore", "types", "workflows"

(syntax-check)